tests/test_device.py

```diff
@@ -62,8 +63,61 @@ def test_supported(self):
             self.assertTrue(len(ops) > 0)
             self.assertTrue(len(exps) > 0)
 
-            for op in ops.union(exps):
-                self.assertTrue(dev.supported(op))
+            for op in ops:
+                self.assertTrue(dev.supports_operation(op))
+
+            for obs in exps:
+                self.assertTrue(dev.supports_observable(obs))
+
+    @patch.multiple(Device, __abstractmethods__=set(), operations=PropertyMock(return_value=['PauliX']))
+    def test_supports_operation_argument_types(self):
+        """check that a the different argument types for the function
+           device.supports_operation are supported"""
+        self.logTestName()
+
+        mock_device = Device()
+
+        self.assertTrue(mock_device.supports_operation('PauliX'))
+        self.assertTrue(mock_device.supports_operation(qml.PauliX))
```

Comment:
Nice!

Reply:
Yeah, I also like that way of doing it! 

I think there are actually plenty of places in the tests that actually feel more like a test of default.qubit than a test of the underlying logic in the core...